File: src/main/java/sk/matusko/wixit/authentication/repository/UserRepository.java
package sk.matusko.wixit.authentication.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.PagingAndSortingRepository;
import sk.matusko.wixit.authentication.dao.User;

public interface UserRepository extends PagingAndSortingRepository<User, Long> {

    Page<User> findByLastName(String lastName, Pageable pageable);

}

File: src/main/kotlin/sk/matusko/wixit/auth/repository/UserRepository.kt
package sk.matusko.wixit.auth.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import sk.matusko.wixit.common.dao.User

interface UserRepository : PagingAndSortingRepository<User, Long> {

    fun findByLastName(lastName: String, pageable: Pageable): Page<User>

    fun findOneByUsername(username: String): User

}